lib/Models/CatalogItem.js

```diff
@@ -876,6 +876,22 @@ CatalogItem.prototype.useClock = function() {
     });
 };
 
+/**
+ * Updates the current time for the catalog item.
+ * This function updates the correct clock (terria.clock or this.clock) depending on whether .useOwnClock is true or
+ * false so that the catalog items state will reflect the new time correctly.
+ *
+ * @param {JulianDate} The new time to update the current time to.
+ */
+CatalogItem.prototype.updateCurrentTime = function(newTime) {
```

Comment:
Could this be the setter on the `currentTime` property, rather than a function?

Reply:
Oh yeah of course, I completely forgot about that when I wrote this, I will fix this now. Thanks for picking this up.